pkgs/build-support/binary-cache.nix

```diff
@@ -0,0 +1,77 @@
+{ stdenv, buildPackages }:
+
+# This function is for creating a flat-file binary cache, i.e. the kind created by
+# nix copy --to file:///some/path and usable as a substituter (with the file:// prefix).
+
+# For example, in the Nixpkgs repo:
+# nix-build -E 'with import ./. {}; mkBinaryCache { rootPaths = [hello]; }'
+
+{ name ? "binary-cache"
+, rootPaths
+}:
+
+stdenv.mkDerivation {
+  inherit name;
+
+  __structuredAttrs = true;
+
+  exportReferencesGraph.closure = rootPaths;
+
+  preferLocalBuild = true;
+
+  PATH = "${buildPackages.coreutils}/bin:${buildPackages.jq}/bin:${buildPackages.python3}/bin:${buildPackages.nix}/bin:${buildPackages.xz}/bin";
+
+  builder = builtins.toFile "builder" ''
```

Comment:
If you're replacing `builder`, why use `stdenv.mkDerivation` rather than `derivation`?

Is replacing `stdenv.mkDerivation`'s `builder` a supported use case?

Reply:
I guess `closure-info.nix` does this too.

Reply:
I assumed `closure-info.nix` has its reasons (minimizing the dependencies somehow?)